lib/broccoli/translation-reducer/utils/wrap-with-namespace-if-needed.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-env node */
+
+'use strict';
+
+const path = require('path');
+
+/**
+ * Wraps the root object with a namespace if the file is under subdirectories
+ *
+ * If a file is under "translations/foo/bar/en-us.json" and we have the "baz" key in that file
+ * we will need to access to the translation using the "foo.bar.baz" key instead of "baz".
+ *
+ * @method wrapWithNamespaceIfNeeded
+ * @param {Object} object
+ * @param {String} filepath
+ * @param {String} inputPath
+ * @param {String[]} addonNames Names of the addons with translations
+ * @return {Object} Returns the input object
+ * @private
+ */
+function wrapWithNamespaceIfNeeded(object, filepath, inputPath, addonNames) {
+  filepath = path.dirname(filepath);
```

Comment:
nit: can we use another variable instead of reassigning the passed in argument

Reply:
Done!
https://github.com/ember-intl/ember-intl/pull/945/commits/04b5dfcd4279acecace102624ade373ea9696bc0